src/InstrumentationEngine.Lib/EnumeratorImpl.h

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "DataContainer.h"
+
+using namespace ATL;
+
+namespace MicrosoftInstrumentationEngine
+{
+    template<class ENUM_INTERFACE, class ENUM_ITEM_TYPE>
+    class CEnumerator : public ENUM_INTERFACE, public CDataContainer
```

Comment:
>CEnumerator [](start = 10, length = 11)

Are there some missing deletes from the PR? (such as this file)

Reply:
Good catch. Looks like I copied instead of moved them.